Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -0,0 +1,1151 @@
+            if log_time := log.get(created_time_field):
+                log['_time'] = log_time
+
+        return logs, new_next_link
+
+    def get_workbench_logs(
+        self,
+        start_datetime: str,
+        end_datetime: str | None = None,
+        order_by: str = 'createdDateTime asc',
+        limit: int = DEFAULT_MAX_LIMIT
+    ) -> List[dict]:
+        """
+        Get the workbench logs.
+
+        docs:
+        https://automation.trendmicro.com/xdr/api-v3#tag/Workbench
+
+        Args:
+            start_datetime (str): Datetime in ISO 8601 format (yyyy-MM-ddThh:mm:ssZ in UTC) that indicates the start
+                                  of the data retrieval time range.
+            end_datetime (str): Datetime in ISO 8601 format (yyyy-MM-ddThh:mm:ssZ in UTC) that indicates the end
+                                of the data retrieval time range.
+            order_by (str): Parameter to be used for sorting records. Records are returned in descending
+                            order by default.
+            limit (int): the maximum number of workbench events to retrieve.
+
+        Returns:
+            List[Dict]: The workbench events that were found.
+        """
+        # will retrieve all the events that are more or equal to start_datetime, does not support miliseconds
+        params = {'startDateTime': start_datetime, 'orderBy': order_by}
+
+        if end_datetime:
+            params['endDateTime'] = end_datetime
+
+        workbench_logs, _ = self.get_logs(
+            url_suffix=UrlSuffixes.WORKBENCH.value,
+            params=params,
+            limit=limit
+        )
+
+        return workbench_logs
+
+    def get_observed_attack_techniques_logs(
+        self,
+        detected_start_datetime: str = '',
+        detected_end_datetime: str = '',
+        top: int = 1000,
+        limit: int = DEFAULT_MAX_LIMIT,
+        next_link: str | None = None
+    ) -> tuple[List[dict], str | None]:
+        """
+        Get the observed attack techniques logs.
+
+        docs:
+        https://automation.trendmicro.com/xdr/api-v3#tag/Observed-Attack-Techniques/paths/~1v3.0~1oat~1detections/get
+
+        Note: The data retrieval time range cannot be greater than 365 days.
+
+        Args:
+            detected_start_datetime (str): Timestamp in ISO 8601 format that indicates the start of the event detection
+                                           data retrieval time range. If no value is specified, detectedStartDateTime
+                                           defaults to 1 hour before the time the request is made.
+            detected_end_datetime (str): Timestamp in ISO 8601 format that indicates the end of the event
+                                         detection data retrieval time range. If no value is specified,
+                                         detectedEndDateTime defaults to the time the request is made.
+            top (int): Number of records displayed on a single page.
```

Comment:
whats the difference between top and limit?

Reply:
top = the number of records that will be returned for each page
limit = the maximum number of events to fetch